views/reset.go

```diff
@@ -0,0 +1,137 @@
+package views
+
+import (
+	"fmt"
+	"github.com/google/uuid"
+	"github.com/labstack/echo/v4"
+	"github.com/patrickmn/go-cache"
+	"github.com/ystv/web-auth/infrastructure/mail"
+	"github.com/ystv/web-auth/templates"
+	"github.com/ystv/web-auth/user"
+	"gopkg.in/guregu/null.v4"
+	"log"
+	"net/http"
+	"strconv"
+)
+
+func (v *Views) ResetURLFunc(c echo.Context) error {
+	url := c.Param("url")
+
+	userID, found := v.cache.Get(url)
+	if !found {
+		return echo.NewHTTPError(http.StatusInternalServerError, fmt.Errorf("failed to get url for reset"))
+	}
+
+	user1, err := v.user.GetUser(c.Request().Context(), user.User{UserID: userID.(int)})
+	if err != nil {
+		v.cache.Delete(url)
+		return echo.NewHTTPError(http.StatusInternalServerError, fmt.Errorf("url is invalid, failed to get user : %w", err))
+	}
+
+	switch c.Request().Method {
+	case "GET":
+		return v.template.RenderTemplate(c.Response(), nil, templates.ResetTemplate, templates.NoNavType)
+	case "POST":
+		err = c.Request().ParseForm()
+		if err != nil {
+			return echo.NewHTTPError(http.StatusInternalServerError, fmt.Errorf("failed to parse form for reset: %w", err))
+		}
+
+		password := c.FormValue("password")
+		if password != c.FormValue("confirmpassword") {
+			return v.template.RenderTemplate(c.Response(), nil, templates.ResetTemplate, templates.NoNavType)
+		}
+
+		err = c.Request().ParseForm()
+		if err != nil {
+			http.Error(c.Response(), err.Error(), http.StatusBadRequest)
+		}
+		p := c.Request().Form.Get("password")
+		if p != c.Request().Form.Get("confirmpassword") || p == "" {
+			return v.template.RenderTemplate(c.Response(), nil, templates.ResetTemplate, templates.NoNavType)
+		}
+		user1.Password = null.StringFrom(password)
+
+		user2, err := v.user.UpdateUserPassword(c.Request().Context(), user1)
```

Comment:
What's the difference between `user1` and `user2` (and could the variables be renamed so I don't have to ask)?

Reply:
Fixed